system/jlib/jutil.hpp

```diff
@@ -360,10 +360,18 @@ extern jlib_decl bool getConfigurationDirectory(const IPropertyTree *dirtree, //
                                                 const char *instance, 
                                                 StringBuffer &dirout);
 
-extern jlib_decl bool querySecuritySettings(bool *          _useSSL,
+extern jlib_decl bool querySecuritySettings(SSLCfg *        _useSSL,
```

Comment:
If useSSL is != SSLNone, then there must be a privatekey + passPhrase right?

Do you need useSSL ?
You could just return insecure port and/or secure port + other fields dependent on configuration.